report/3.Theory/Parser.tex

```diff
@@ -0,0 +1,112 @@
+\section{Parser}
+
+The second part of the compiler will parse the lexical tokens coming from the lexer. 
+This part of the compiler is called the parser. 
+The parser works by taking a sequence of lexical tokens, and generating a parse tree or transforming them into an abstract syntax tree. 
+
+\subsection{Output data types}
+
+The parser will output some data type, that represents the compiled program in a hierarchical structure. 
+The outputted data type should hold all the information or almost all the information needed to generate the code later. 
+Sometimes the data type will be accommodated by a symbol table in order to generate the desired code.
+
+\textbf{Parse tree}
+
+One of the possible data structures to be outputted from the parser is the parse tree explained in section \ref{CFGdescription}. 
+The parse tree should resemble the structure of the CFG and the concrete input to the parser. 
+The parse tree is easily generated, and often the actual output from the parser. 
+However, the parse tree contains a lot of excessive information, which is not needed to understand the complete meaning of the program. 
+This means that a parse tree is both hard to analyze and slow to traverse. 
+Therefore, another common output type from the parser is the abstract syntax tree.
+
+\textbf{Abstract syntax tree}
+
+The abstract syntax tree is a sparse version of the parse tree. 
+It still contains all the information needed to understand the structure of the program, but in a more compressed way. 
+For instance parentheses can be omitted in the abstract syntax tree.
+
+Recall the parse tree from figure \ref{fig:AmbiguousGrammarEx1} a corresponding abstract syntax tree can bee seen on figure \ref{fig:ast} 
```

Comment:
Think you're missing a . after the first reference (and if so, obviously caps the first "a")